Added updateOrCreate function

makoru-hikage opened the pull request:
A response to Issue #107. Its arguments are the same as the arguments of Repofuck's `update()`

rmrhz committed:
Merge pull request #102 from rmrhz/master

Fix unexpected thrown exception

makoru-hikage committed:
Correcting the premature end of a loop

rmrhz committed:
Merge pull request #103 from makoru-hikage/master

Correcting the premature end of a loop

makoru-hikage committed:
Fixing inconsistencies: find() functions must use  as arguments

rmrhz committed:
Merge pull request #106 from makoru-hikage/master

Fixed inconsistencies: find() functions must use $columns as arguments

makoru-hikage committed:
Added updateOrCreate function

rmrhz commented on src/Prjkt/Component/Repofuck/Repofuck.php:
```diff
@@ -450,6 +450,26 @@ public function update($identifier) : Model
+		$entity = $identifier;
+
+		if (! $entity instanceof Model) {
+			$entity = $this->first($identifier) ?? new $this->entity;
```
This is an incomplete implementation. Instead of using the `$identifier` provided over the parameter. Check the `$data` property if the identifier exists.

rmrhz commented:
@makoru-hikage please respond on review

makoru-hikage commented:
I can't find the review

rmrhz reviewed (commented):
Please fix the following

makoru-hikage committed:
UpdateOrCreate update...

Eloquent Model's `updateOrCreate` also uses `firstOrNew`. It's like the `Model`'s `updateOrCreate`, but integrated with `Repofuck`'s `map()`.

makoru-hikage committed:
Update Repofuck.php

makoru-hikage committed:
Update Repofuck.php

rmrhz commented on src/Prjkt/Component/Repofuck/Repofuck.php:
```diff
@@ -452,17 +452,32 @@ public function update($identifier) : Model
-			$entity = $this->first($identifier) ?? new $this->entity;
+		switch ($identifier) 
+		{
+			case null:
```
Instead of null, change it to `default`.

rmrhz reviewed (commented):
More changes.

makoru-hikage committed:
Changed `case null` to `default`

makoru-hikage committed:
Update Repofuck.php

rmrhz commented on src/Prjkt/Component/Repofuck/Repofuck.php:
```diff
@@ -399,11 +399,13 @@ public function prepare(Closure $function) : self
 	 * @return \Illuminate\Database\Eloquent\Collection
 	 */
-	public function get() : Collection
+	public function get($columns = null) : Collection
```
Instead of defining columns here, why not use the `columns` method?

rmrhz reviewed (commented):
@makoru-hikage see review

makoru-hikage committed:
Must set columns through the `columns()` method
